nestjs/src/courses/task-verifications/self-education.service.ts

```diff
@@ -52,6 +46,41 @@ export class SelfEducationService {
     studentId,
     studentAnswers,
   }: SelfEducationVerificationParams) {
+    const { id: courseTaskId, type: courseTaskType } = courseTask;
+
+    const verifications = await this.taskVerificationsRepository.find({
+      where: { studentId, courseTaskId },
+      order: { createdDate: 'DESC' },
+    });
+
+    const { score, details, studentCorrectAnswers } = this.verifySelfEducationAnswers(
+      courseTask,
+      studentAnswers,
+      verifications.length,
+      verifications[0]?.createdDate?.toString(),
+    );
+
+    const { id } = await this.taskVerificationsRepository.save({
+      studentId,
+      courseTaskId,
+      score,
+      details,
+      status: 'completed',
+      answers: studentCorrectAnswers,
+    });
+
+    const result = await this.taskVerificationsRepository.findOneByOrFail({ id });
+    await this.writeScoreService.saveScore(studentId, courseTaskId, { score, comment: details });
```

Comment:
Minor:  
```suggestion
    const [result] = await Promise.all([
      this.taskVerificationsRepository.findOneByOrFail({ id }),
      this.writeScoreService.saveScore(studentId, courseTaskId, { score, comment: details }),
    ]);
```

Reply:
since "saveScore" is "complex" and requires additional refactoring, I would leave as is. It's not critical for performance and it's better to execute in the order